Tests/GRPCHTTP2CoreTests/Client/Connection/RequestQueueTests.swift

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import XCTest
+
+@testable import GRPCHTTP2Core
+
+@available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
+final class RequestQueueTests: XCTestCase {
+  struct AnErrorToAvoidALeak: Error {}
+
+  func testPopFirstEmpty() {
+    var queue = RequestQueue()
+    XCTAssertNil(queue.popFirst())
+  }
+
+  func testPopFirstNonEmpty() async {
+    _ = try? await withCheckedThrowingContinuation { continuation in
+      var queue = RequestQueue()
+      let id = QueueEntryID()
+
+      queue.append(continuation: continuation, waitForReady: false, id: id)
+      guard let popped = queue.popFirst() else {
+        return XCTFail("Missing continuation")
+      }
+      XCTAssertNil(queue.popFirst())
+
+      popped.resume(throwing: AnErrorToAvoidALeak())
+    }
+  }
+
+  func testRemoveEntryByID() async {
```

Comment:
Can we add/change this test where we have more than a single entry and we get one of them by ID?